translator/test/src/test/resources/features/TranslatorUnitTests.feature

```diff
@@ -152,4 +152,171 @@ Feature: Translator Service
+
+  Scenario: Translating invalid jms data message with valid channel, body and metrics into kura data message
+  Creating null jms message with valid channel, body and metrics. Trying to translate it into invalid jms message.
+  Check if mqtt message with valid topic and encoded body is received.
+  Invalid message exception should be thrown
+
+    Given I create jms message with valid payload "response.code" and valid topic "kapua-sys.rpione3.DEPLOY-V2.GET.packages"
+    Then I expect the exception "InvalidMessageException" with the text "Invalid message: null"
+    When I try to translate invalid jms message to kura data message
+    And An exception was thrown
+
+  #TranslatorDataKuraJms
+
+  Scenario: Translating kura data message with valid channel and without body and metrics into jms message
+  Creating kura data message with valid channel and without body and metrics. Trying to translate it into jms message.
+  Check if jms message with valid topic and empty payload body is received.
+
+    Given I create kura data message with channel with scope "kapua-sys", client id "rpione3" and payload without body and metrics
+    When I try to translate kura data message to jms message
+    Then I got jms message with topic "kapua-sys.rpione3" and empty body
+    And No exception was thrown
+
+  Scenario: Translating kura data message with valid channel, metrics and without body into jms message
+  Creating kura data message with valid channel, metrics and without body. Trying to translate it into jms message.
+  Check if jms message with valid topic and encoded payload body is received.
+
+    Given I create kura data message with channel with scope "kapua-sys", client id "rpione3" and payload without body and metrics
+    When I try to translate kura data message to jms message
+    Then I got jms message with topic "kapua-sys.rpione3" and non empty body
+    And No exception was thrown
+
+  Scenario: Translating kura data message with valid channel, body and metrics into jms message
+  Creating kura data message with valid channel, body and metrics. Trying to translate it into jms message.
+  Check if jms message with valid topic and encoded payload body is received.
+
+    Given I create kura data message with channel with scope "kapua-sys", client id "rpione3" and payload with body and metrics
+    When I try to translate kura data message to jms message
+    Then I got jms message with topic "kapua-sys.rpione3" and non empty body
+    And No exception was thrown
+
+  Scenario: Translating kura data message with valid channel, and with null payload
+  Creating kura data message with valid channel and with "null" payload Trying to translate it into jms message.
+  Invalid payload exception should be thrown.
+
+    Given I create kura data message with channel with scope "kapua-sys", client id "rpione3" and null payload
+    Then I expect the exception "InvalidPayloadException" with the text "Invalid payload: null"
```

Comment:
Maybe this is something I need  to check. 
There should be a meaningful message